### DISCUSSION TITLE
Create a template bb.edn file programmatically/via the CLI

### DISCUSSION THREAD

**Discussion by craftybones** [babashka/babashka#1746] [Ideas]
**Is your feature request related to a problem? Please describe.**
Currently, there is no way to generate a template `bb.edn` file. I have to always look at documentation for the various keys and intricacies. It would be wonderful if there was a way to generate a template either directly from the command line or programmatically from the REPL(or both)

**Describe the solution you'd like**
I would like babashka to have a subcommand like `init` so that typing in `bb init` would generate a template `bb.edn` file. Additionally I would like this to be invokable from a repl such as `(bb/init-bb-edn)` so that I get an EDN back. Having it available programmatically would be wonderful as I can simply alter the EDN via a REPL and/or merge it with some other config from elsewhere.

**Comment by borkdude** [comment]
One other option might be to add this to neil: https://github.com/babashka/neil

**Comment by bobisageek** [comment]
I was experimenting a bit, and a thought that came to mind was the idea of a couple of deps-new templates. This could open the door for a few variants on the outcome that could be somewhat comparable to the built-ins in deps-new, but aimed at babashka (e.g. 'script' and 'project').

With just a template, `neil new` would be able to support it as-is (with a dep for those templates), and there's maybe a further possibility for `neil` having some specific knowledge of these templates to shorten up the params needed to `neil new` for handling these (maybe babashka org-owned) template(s).

**Comment by borkdude** [comment]
Yes, this can be done now entirely with neil I think. Turning this into a Github Discussion.
